internal/topo/transform/template.go

```diff
@@ -83,8 +119,22 @@
 				}
 				d = m
 			}
-			b, err := c.Encode(d)
-			return b, transformed, err
+			if !s || len(fields) == 0 {
+				b, err := c.Encode(d)
+				if err != nil {
+					return nil, false, fmt.Errorf("fail to encode data %v for error %v", d, err)
+				}
+				return b, true, nil
+			}
+			mm, ok := d.(map[string]interface{})
```

Comment:
d must be map or []map. Should deal with both types; Maybe let selectMap accept d and return interface because Encode accept an interface.